src/Propel/Generator/Model/Table.php

```diff
@@ -784,6 +784,25 @@ public function setupReferrers($throwErrors = false)
                 }
             }
 
+            // check for incomplete foreign key references when foreign table
+            // has a composite primary key
+            if ($foreignTable->hasCompositePrimaryKey()) {
+                // get composite foreign key's keys
+                $foreignPrimaryKeys = $foreignTable->getPrimaryKey();
+                // check all keys are referenced in foreign key
+                foreach ($foreignPrimaryKeys as $foreignPrimaryKey) {
+                    if (!$foreignPrimaryKey->hasReferrer($foreignKey) && $throwErrors) {
```

Comment:
@marcj could you please elaborate the exact situation we are trying to catch here related to unique's ?

According to #675 
> See http://stackoverflow.com/questions/7947085/are-incomplete-key-references-in-propel-useful
> 
> When foreign-key points to a table with composite primary key but we have only defined one reference in our foreign key then we should throw a error.
> 
this pull request fixes exactly this...

I tried to understand/find your unique key issue but I fail to do so :(